server/src/main/java/org/elasticsearch/common/filesystem/FileSystemNatives.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.filesystem;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.Constants;
+import org.elasticsearch.core.Nullable;
+
+import java.nio.file.Path;
+
+/**
+ * This class provides utility methods for calling some native methods related to filesystems.
+ */
+public final class FileSystemNatives {
+
+    private static final Logger logger = LogManager.getLogger(FileSystemNatives.class);
+
+    interface Provider {
+        @Nullable
+        Long allocatedSizeInBytes(Path path);
+    }
```

Comment:
I know that there is history and consistency constraints being applied to this code, but what do you think of dropping Nullable here, and using an Optional instead - to better represent the fact that this support is optional ( depending on platform support ), e.g.

```   
interface Provider {
    OptionalLong allocatedSizeInBytes(Path path);
}
```

Reply:
Makes sense, thanks for the suggestion